Jest can't find Node's 'v8' module

**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
```
Cannot find module 'v8' from 'myfile.js'
```

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal repository on GitHub that we can `npm install` and `npm test`.**
require the `v8` module in a Jest test file.

**What is the expected behavior?**
No error

**Run Jest again with `--debug` and provide the full configuration it prints. Please mention your node and npm version and operating system.**

```
$ node -v
v7.1.0
$ npm -v
3.10.9
jest version = 17.0.2
test framework = jasmine2
config = {
  "verbose": true,
  "setupFiles": [
    "/Users/anmonteiro/Documents/github/lumo/node_modules/babel-polyfill/lib/index.js",
    "/Users/anmonteiro/Documents/github/lumo/scripts/jest/environment.js"
  ],
  "collectCoverageFrom": [
    "src/js/**/*.js"
  ],
  "coverageDirectory": "/Users/anmonteiro/Documents/github/lumo/coverage",
  "coveragePathIgnorePatterns": [
    "/Users/anmonteiro/Documents/github/lumo/lib",
    "/Users/anmonteiro/Documents/github/lumo/node_modules"
  ],
  "testPathDirs": [
    "/Users/anmonteiro/Documents/github/lumo/src/js"
  ],
  "rootDir": "/Users/anmonteiro/Documents/github/lumo",
  "name": "-Users-anmonteiro-Documents-github-lumo",
  "testRunner": "/Users/anmonteiro/Documents/github/lumo/node_modules/jest-jasmine2/build/index.js",
  "transform": [
    [
      "^.+\\.jsx?$",
      "/Users/anmonteiro/Documents/github/lumo/node_modules/babel-jest/build/index.js"
    ]
  ],
  "usesBabelJest": true,
  "automock": false,
  "bail": false,
  "browser": false,
  "cacheDirectory": "/var/folders/d_/b1bghypn20lgq2b24v7396pr0000gn/T/jest",
  "coverageReporters": [
    "json",
    "text",
    "lcov",
    "clover"
  ],
  "expand": false,
  "globals": {},
  "haste": {
    "providesModuleNodeModules": []
  },
  "mocksPattern": "__mocks__",
  "moduleDirectories": [
    "node_modules"
  ],
  "moduleFileExtensions": [
    "js",
    "json",
    "jsx",
    "node"
  ],
  "moduleNameMapper": {},
  "modulePathIgnorePatterns": [],
  "noStackTrace": false,
  "notify": false,
  "preset": null,
  "resetMocks": false,
  "resetModules": false,
  "snapshotSerializers": [],
  "testEnvironment": "jest-environment-jsdom",
  "testPathIgnorePatterns": [
    "/node_modules/"
  ],
  "testRegex": "(/__tests__/.*|\\.(test|spec))\\.jsx?$",
  "testURL": "about:blank",
  "timers": "real",
  "transformIgnorePatterns": [
    "/node_modules/"
  ],
  "useStderr": false,
  "watch": false,
  "cache": true,
  "watchman": true
}
```

--- a/integration_tests/__tests__/require-v8-module-test.js
+++ b/integration_tests/__tests__/require-v8-module-test.js
@@ -1,0 +1,15 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails oncall+jsinfra
+ */
+
+test('v8 module', () => {
+  expect(() => require('v8')).not.toThrow();
+
+  expect(require('v8').getHeapStatistics().total_heap_size).toBeDefined();
+});

--- a/packages/jest-resolve/src/index.js
+++ b/packages/jest-resolve/src/index.js
@@ -156,7 +156,13 @@ class Resolver {
   }
 
   isCoreModule(moduleName: string): boolean {
-    return this._options.hasCoreModules && resolve.isCore(moduleName);
+    return (
+      this._options.hasCoreModules &&
+      (
+        resolve.isCore(moduleName) ||
+        moduleName === 'v8'
+      )
+    );
   }
 
   getModule(name: string): ?Path {

